src/sources/datadog_agent/logs.rs

```diff
@@ -192,3 +201,71 @@ pub(crate) fn decode_log_body(
 
     Ok(decoded)
 }
+
+// ddtags input is a string containing a list of key-value pairs.
+// the tag-value list members are separated by `,` and the
+// tag-value pairs are separated by `:`.
+//
+// The output is an Object constructed with the tag-value pairs.
+fn parse_ddtags(ddtags_raw: &Bytes) -> ObjectMap {
+    let ddtags_str = String::from_utf8_lossy(ddtags_raw);
+
+    let mut ddtags_object = ObjectMap::new();
+
+    ddtags_str.split(',').for_each(|kv| {
+        if let Some((k, v)) = kv.split_once(':') {
+            ddtags_object.insert(KeyString::from(k), Value::Bytes(Bytes::from(v.to_string())));
+        }
+    });
```

Comment:
nit: Can be done entirely within an iterator:
```suggestion
    let ddtags_object: ObjectMap = ddtags_str
        .split(',')
        .flat_map(|kv| kv.split_once(':'))
        .map(|(k, v)| (KeyString::from(k), Value::Bytes(Bytes::from(v.to_string()))))
        .collect;
```
However, I wonder:
1. Have we not done this exact operation before in Vector? Surely this isn't the only thing parsing key-value pairs like this.
2. Are bare tags (no `:`) not valid ddtags as well?

Reply:
> Are bare tags (no :) not valid ddtags as well?

I couldn't find any documentation on this. 

Nor could I find any examples when searching the entire Datadog GH org. All instances had the `:` as a pair.

But. I did a local experiment sending to the logs intake, and using a bare tag. It was accepted with a 202 🤷  so I guess we should support it.

Reply:
> Have we not done this exact operation before in Vector? Surely this isn't the only thing parsing key-value pairs like this.

I did just find `extract_tag_key_and_value()`, a helper function used in the `datadog_agent` source when handling metrics. It operates similarly over colon delimited kv pairs. 

However it is coupled with `MetricTags`, for which I'm not seeing a way to get that into a `Value` without some translation here.

Reply:
I attempted to resolve this in the latest changes.